pkg/transport/transport.go

```diff
@@ -130,6 +138,15 @@ func (transport *Transport) WithObservers(obs ...Observer) Client {
 	return tr
 }
 
+func getIP(target string) (string, error) {
+	addrElements := strings.Split(target, ":")
+	ipAddr, err := net.ResolveIPAddr("ip", addrElements[0])
```

Comment:
Do we have any DNS redirect loop concerns here?

Reply:
I asked about this during standup. @thepaul said the go resolver might take care of this, so JT said we might not be concerned

Reply:
Would it be feasible to write a test case for this?